### PR TITLE
improve comments and relax one cmpxch in AtomicWaker

### PR DESCRIPTION
**Pull request by rkuhn**
While reading AtomicWaker I stumbled over some comments that were pointing slightly off path, so I’m offering potential improvements in case my understanding of the code is correct.

Caveat emptor: I have only reasoned about the code, not tested it. If such code-based discussion is not common in the Rust community then please let me know where I should raise this point instead. I’m looking forward to learning from your responses!

### CODE REVIEW

File: futures-core/src/task/__internal/atomic_waker.rs [hunk 1]
```diff
@@ -268,15 +268,18 @@ impl AtomicWaker {
...
+                    // 
                     // Start by assuming that the state is `REGISTERING` as this
-                    // is what we jut set it to.
+                    // is what we just set it to. If this holds, we know that no
+                    // other writes were performed in the meantime, so there is
+                    // nothing to acquire, but release the update. In case of
+                    // concurrent wakers, we need to acquire their releases.
                     let res = self.state.compare_exchange(
-                        REGISTERING, WAITING, AcqRel, Acquire);
+                        REGISTERING, WAITING, Release, Acquire);
```
Comment (cramertj) [inline futures-core/src/task/__internal/atomic_waker.rs]: Errors with `a failure ordering can't be stronger than a success ordering`

Comment (rkuhn) [reply futures-core/src/task/__internal/atomic_waker.rs]: Ah, yes, makes sense, will update.

**Comment by rkuhn** [comment]
Thanks for the review @cramertj! This code is so well documented that I think it makes sense to add the now remaining comment clarifications — it can serve as a tutorial for how to do wait-free synchronization in Rust :-)

**Comment by cramertj** [comment]
Thanks so much!

**Pull request merged**